js/index.js

```diff
@@ -1,274 +1,158 @@
-    g6Board.addEventListener("click", function(event) {
-    });
+/*
+Add functionality to the event listeners for when 
+a piece is clicked on the chessboard.
+*/
+function initBoardEvListener() {
+    const rows = 8;
+    const cols = ["a", "b", "c", "d", "e", "f", "g", "h"];
+
+    for (let i = 1; i <= rows; i++) {
+        for (let j = 0; j < cols.length; j++) {
+            const squareString = cols[j] + i.toString();
+            const element = document.getElementById(squareString);
+            element.addEventListener("click", () => {
+                handleSquareClick(squareString);
+            });
 
-    const g7Board = document.getElementById("g7");
-    g7Board.addEventListener("click", function(event) {
-    });
+        }
 
-    const g8Board = document.getElementById("g8");
-    g8Board.addEventListener("click", function(event) {
-    });
+    }
+}
 
-    // First column. 'h' column
-    const h1Board = document.getElementById("h1");
-    h1Board.addEventListener("click", function(event) {
-    });
+/*
+Function to mark the possible moves on the chessboard that 
+a player can move to. 
+*/
+function handleSquareClick(squareString) {
+    const piece = getPiece(squareString); //get the Piece on the square 
+    if (piece && piece.getPlayerId() == currentPlayer) { //if piece belongs to current player 
+        activePiece = piece; //sets activePiece to piece
+        colorAllSquares(); //resets the colors of the squares
+        const potentialMoves = piece.possibleMoves(board.board); //potential moves that the piece can go
+        board.markPossibleMoves(potentialMoves); //Marking the possible squares a piece can move to 
+        const markedSquares = document.getElementsByClassName("markedSquare");
+        for (let i = 0; i < markedSquares.length; i++) {
+            markedSquares[i].addEventListener("click", handleMoveToSquare);
+        }
+    }
+}
 
-    const h2Board = document.getElementById("h2");
-    h2Board.addEventListener("click", function(event) {
-    });
 
-    const h3Board = document.getElementById("h3");
-    h3Board.addEventListener("click", function(event) {
-    });
+/*
+Function to mark the possible moves on the chessboard that 
+a player can move to. 
+*/
+function handleMoveToSquare(event) {
+    //get marked square that was clicked 
+    const clickedSquare = event.target; 
+    const clickedSquareId = clickedSquare.id; 
+    const [row, col] = squareStringConverter(clickedSquareId); 
+    //move the piece, update board, reset board colors and active piece variable.
+    activePiece.movePiece(row, col, board);
+    board.renderPieces(); 
+    colorAllSquares(); 
+    activePiece = null; 
+    //remove event listener here 
+    const markedSquares = document.getElementsByClassName("markedSquare");
+    for (let i = 0; i < markedSquares.length; i++) {
+        markedSquares[i].removeEventListener("click", handleMoveToSquare);
+        markedSquares[i].classList.remove("markedSquare");
+    }
```

Comment:
Removing the event listeners on lines `97 - 102` still allows pieces to move to the marked squares after they are no longer marked since the `handleMoveToSquare()` event listeners will fire *before* the event listeners are removed and move the piece.

> You might want to consider turning the `if` statement on line `70` into an `if-else-if` statement that handles the move functionality instead of adding additional event listeners to make it easier on yourself.